### DISCUSSION TITLE
Volume detachment while node failure

### DISCUSSION THREAD

**Discussion by PlatformLC** [longhorn/longhorn#2943] [Q&A]
Hi Experts,
   I'm using k8s 1.18.0 and longhorn 1.1.2 While testing node failure case with  setting `nodeDownPodDeletionPolicy` to `delete-both-statefulset-and-deployment-pod`. I find that  longhorn won't detach volume from the failure node initiatively, but we must wait until k8s hardcoded 6 mins detach timeout to forcely detach the corresponding volume, then the new Pods could attach the volume successfully.
It seems we intendedly remove "Volume Attachment Recovery Policy feature" from 1.1.2 (https://github.com/longhorn/longhorn/issues/2060), is the result I tested expected from the design? Or something I missed or misconfigured?
But before 1.1.2, we could set `volumeAttachmentRecoveryPolicy` to `immediate`/`wait` to detach the volume initiatedly, then the new scheduled Pods attach the volume rapidly.

Looking forward for your reply!

**Comment by c3y1huang** [comment]
/ping @joshimoo , could you help to give more explanation about this one?

**Reply by PlatformLC** [reply]
Thanks @c3y1huang for looking into the query. @joshimoo Any information or clues for it? Looking forward for your reply.
